File: cql-ls-plugin-debug/src/test/java/org/opencds/cqf/cql/ls/plugin/debug/client/TestDebugClient.java
package org.opencds.cqf.cql.ls.plugin.debug.client;

import java.util.concurrent.CompletableFuture;

import org.eclipse.lsp4j.debug.ExitedEventArguments;
import org.eclipse.lsp4j.debug.OutputEventArguments;
import org.eclipse.lsp4j.debug.services.IDebugProtocolClient;

public class TestDebugClient implements IDebugProtocolClient {
    private String serverOutput = null;

    private CompletableFuture<Void> exited;
    public TestDebugClient() {
        this.exited = new CompletableFuture<>();
    }

    public String getServerOutput() {
        return this.serverOutput;
    }

    @Override
    public void initialized() {
    }

    @Override
	public void output(OutputEventArguments args) {
        // this.serverOutput = args.getOutput();

	}

    @Override
    public void exited(ExitedEventArguments args){
        this.serverOutput = "got exited";
        this.exited.complete(null);
    }

    public CompletableFuture<Void> exited() {
        return this.exited;
    }
}

File: plugin/debug/src/test/java/org/opencds/cqf/cql/ls/plugin/debug/client/TestDebugClient.kt
package org.opencds.cqf.cql.ls.plugin.debug.client

import org.eclipse.lsp4j.debug.ExitedEventArguments
import org.eclipse.lsp4j.debug.OutputEventArguments
import org.eclipse.lsp4j.debug.services.IDebugProtocolClient
import java.util.concurrent.CompletableFuture

class TestDebugClient : IDebugProtocolClient {
    private var serverOutput: String? = null

    private val exitedFuture: CompletableFuture<Void> = CompletableFuture()

    fun getServerOutput(): String? {
        return this.serverOutput
    }

    override fun initialized() {}

    override fun output(args: OutputEventArguments) {
        // this.serverOutput = args.output
    }

    override fun exited(args: ExitedEventArguments) {
        this.serverOutput = "got exited"
        this.exitedFuture.complete(null)
    }

    fun exited(): CompletableFuture<Void> {
        return this.exitedFuture
    }
}
